django/db/backends/postgresql/base.py

```diff
@@ -144,6 +145,36 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self._named_cursor_idx = 0
 
+    @classmethod
+    def config_from_url(cls, engine, scheme, url):
+        parsed = parse.urlparse(url)
```

Comment:
In the base class's method, it checks whether `URL` is already parsed. Does this class need to do the same check before calling urlparse?

Reply:
See also the SQLite backend.

Reply:
It's a bit iffy, but I added the check to only the base class so that if the method was over-ridden and the `url` was parsed the result could be passed in without needing to re-call `urlparse`. `config_from_url`, when over-ridden, is always going to parse the URL, but the public API should expect only strings to be passed in.

Reply:
What if someone wants to subclass the Postgres backend directly?

Reply:
Hmm. :<